javafx-example/src/main/java/AnswerService.java

```diff
@@ -0,0 +1,32 @@
+import dev.langchain4j.model.openai.OpenAiStreamingChatModel;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import static java.time.Duration.ofSeconds;
+
+public class AnswerService {
+
+    private static final Logger LOGGER = LogManager.getLogger(AnswerService.class);
+
+    private OpenAiStreamingChatModel model;
+
+    public void init(SearchAction action) {
+        action.appendAnswer("Initiating...");
+        initChat(action);
+    }
+
+    private void initChat(SearchAction action) {
+        model = OpenAiStreamingChatModel.builder()
+                .apiKey(ApiKeys.OPENAI_API_KEY)
+                .timeout(ofSeconds(60))
+                .build();
+        action.appendAnswer("Done");
+        action.setFinished();
+    }
+
+    void ask(SearchAction action) {
+        LOGGER.info("Asking question '" + action.getQuestion() + "'");
+
+        model.generate(action.getQuestion(), new CustomStreamingResponseHandler(action));
```

Comment:
You could probably use `AiServices` with `ChatMemory` (see `ServiceWithMemoryExample`) or `ConversationalChain` (see `ConversationalChain_Example`) to keep track of the conversation. Now every question is standalone and LLM does not remember any previous questions.

Reply:
I tried to use a memory implementation, but I'm struggling a bit how to get the answer as a stream.

These methods don't accept a handler:

```
chain.execute(action.getQuestion()) 
 but need --> chain.execute(action.getQuestion(), new CustomStreamingResponseHandler(action));

or

assistant.chat(action.getQuestion()) 
 but need --> assistant.chat(action.getQuestion(), new CustomStreamingResponseHandler(action));
```

Reply:
Please check `ServiceWithStreamingExample`, you can do something like this:

        StreamingChatLanguageModel model = OpenAiStreamingChatModel.withApiKey(System.getenv("OPENAI_API_KEY"));

        Assistant assistant = AiServices.builder(Assistant.class)
                .streamingChatLanguageModel(model)
                .chatMemory(MessageWindowChatMemory.withMaxMessages(10))
                .build();

Reply:
Then this works:
```
        assistant.chat("My name is Klaus")
                .onNext(System.out::println)
                .onComplete(System.out::println)
                .onError(Throwable::printStackTrace)
                .start();

        Thread.sleep(2000);
        
        assistant.chat("What is my name?")
                .onNext(System.out::println)
                .onComplete(System.out::println)
                .onError(Throwable::printStackTrace)
                .start();
```

Reply:
Forgot the most important part (`TokenStream`):

```
    interface Assistant {

        TokenStream chat(String message);
    }
```

Reply:
Thanks and working perfectly :-)

See the updated screenshot...